### PR TITLE
don't use webhook config for incoming webhook url

### PR DESCRIPTION
**Pull request by rcerven**
[STONEBLD-2681](https://issues.redhat.com//browse/STONEBLD-2681)

### Checklist:
 - PR has reference to the issue(s) it resolves
 - Write / update unit tests
 - Write / update integration (envtest) tests
 - Ensure there is an issue for e2e tests if needed
 - Ensure `make test` passes
 - Ensure test coverage hasn't decreased
 - Test code changes manually
 - Update readme and documentation
 - Write PR description that highlights overall changes and add usage examples if applicable

### CODE REVIEW

File: controllers/component_build_controller_pac.go [hunk 1]
```diff
@@ -373,10 +373,10 @@ func (r *ComponentBuildReconciler) UndoPaCProvisionForComponent(ctx context.Cont
 }
 
 // getPaCWebhookTargetUrl returns URL to which events from git repository should be sent.
-func (r *ComponentBuildReconciler) getPaCWebhookTargetUrl(ctx context.Context, repositoryURL string) (string, error) {
+func (r *ComponentBuildReconciler) getPaCWebhookTargetUrl(ctx context.Context, repositoryURL string, checkConfig bool) (string, error) {
 	webhookTargetUrl := os.Getenv(pipelinesAsCodeRouteEnvVar)
 
-	if webhookTargetUrl == "" {
+	if webhookTargetUrl == "" && checkConfig {
```
Comment (mmorhun) [inline controllers/component_build_controller_pac.go:379]: We can more meaningful name for `checkConfig`

File: controllers/component_build_controller_pac.go [hunk 2]
```diff
@@ -373,10 +373,10 @@ func (r *ComponentBuildReconciler) UndoPaCProvisionForComponent(ctx context.Cont
 }
 
 // getPaCWebhookTargetUrl returns URL to which events from git repository should be sent.
-func (r *ComponentBuildReconciler) getPaCWebhookTargetUrl(ctx context.Context, repositoryURL string) (string, error) {
+func (r *ComponentBuildReconciler) getPaCWebhookTargetUrl(ctx context.Context, repositoryURL string, checkConfig bool) (string, error) {
```
Comment (mmorhun) [inline controllers/component_build_controller_pac.go:376]: Please update the comment with detailed description.

File: controllers/component_build_controller_pac.go (hunk 1 above)
Comment (rcerven) [reply controllers/component_build_controller_pac.go]: changed

File: controllers/component_build_controller_pac.go (hunk 2 above)
Comment (rcerven) [reply controllers/component_build_controller_pac.go]: added

**Review by mmorhun — APPROVED** [review]

**Pull request merged**